### PR TITLE
Cloudinary post with image

### PR DESCRIPTION
**Pull request by lorencemt31**
Reforked to get latest updates and recheck the task. Done with some additional adjustments.

### CODE REVIEW

File: src/app/shared/components/post-textarea/post-textarea.component.html [hunk 1]
```diff
@@ -5,16 +5,13 @@
...
-    </textarea>
-    <input class="upload-file" id="uploadImage" type="file" style="display:none;"/>
+    <textarea class="form-control" placeholder="Whats happening?" name="post_title" [(ngModel)]="postMessage"></textarea>
+
+    <shared-upload-image-component *ngIf="toggleUploadComponent"></shared-upload-image-component>
+
     <div class="text-right options">
-      <label for="uploadImage" class="img-upload flat-btn"> <i class="material-icons">link</i> </label>
-      <input type="submit" class="btn btn-blue btn-sm post-btn" value="ADD POST">
+      <label class="img-upload flat-btn" (click)="showUploadComponent()"> <i class="material-icons">image</i> </label>
```
Comment (caninojories) [inline src/app/shared/components/post-textarea/post-textarea.component.html]: @rence sama click events basically events sa angular gawin natin na may 'on' example onShowUploadComponent(); basically lahat ng mga events sample of event is (click)

File: src/app/shared/components/post-textarea/post-textarea.component.html [hunk 2]
```diff
@@ -5,16 +5,13 @@
...
+      <input type="submit" class="btn btn-blue btn-sm post-btn" value="ADD POST" (click)="addPost()">
```
Comment (caninojories) [inline src/app/shared/components/post-textarea/post-textarea.component.html]: this one also

File: src/app/shared/components/post-textarea/post-textarea.component.html (hunk 1 above)
Comment (lorencemt31) [reply src/app/shared/components/post-textarea/post-textarea.component.html]: okay sir.

File: src/app/shared/components/post-textarea/post-textarea.component.html [hunk 3]
```diff
@@ -5,16 +5,13 @@
     <li role="presentation"><a href="#">Share a Story</a></li>
   </div>
   <div class="post">
-    <textarea
-      class="form-control"
-      placeholder="Whats happening?"
-      name="post_title"
-      [(ngModel)]="postMessage">
```
Comment (caninojories) [inline src/app/shared/components/post-textarea/post-textarea.component.html]: use the PostModel and basically assign the key message ex. protected Post: PostModel = new PostModel();
Then use the post.message as the ngModel

File: src/app/shared/components/post-textarea/post-textarea.component.html (hunk 2 above)
Comment (lorencemt31) [reply src/app/shared/components/post-textarea/post-textarea.component.html]: noted

File: src/app/shared/components/post-textarea/post-textarea.component.html [hunk 4]
```diff
@@ -5,16 +5,13 @@
...
-    </textarea>
-    <input class="upload-file" id="uploadImage" type="file" style="display:none;"/>
+    <textarea class="form-control" placeholder="Whats happening?" name="post_title" [(ngModel)]="postMessage"></textarea>
```
Comment (caninojories) [inline src/app/shared/components/post-textarea/post-textarea.component.html]: use the PostModel and basically assign the key message ex. protected Post: PostModel = new PostModel();
Then use the post.message as the ngModel

File: src/app/shared/components/upload-image/upload-image.component.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<div class="shared-upload-image-component">
+  <div class="file-drop-container" ng2FileDrop [uploader]="uploader" (fileOver)="fileOverBase($event)" [ngClass]="{'nv-file-over': hasBaseDropZoneOver}">
```
Comment (caninojories) [inline src/app/shared/components/upload-image/upload-image.component.html]: this one also onFileOverBase

File: src/app/shared/components/upload-image/upload-image.component.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    <form>
+      <input class="upload-file" id="uploadImage" accept="image/*" ng2FileSelect #fileInput type="file" style="display:none;" multiple [uploader]="uploader" (change)="fileInput.value=''" accept="image/gif, image/jpeg, image/png" />
+    </form>
+    <div class="file-drop-or-select">
+      Add images by dropping them here... <br />
+      <em> Maximum of 4 images only. </em>
+    </div>
+    <div class="img-to-upload-list" *ngIf="imagesToUpload">
+      <div class="img-crop" *ngFor="let imageUrl of imagesToUpload; let i = index;">
+        <div class="remove-img" (click)="removeFromQueue(i)"><i class="material-icons">close</i></div>
```
Comment (caninojories) [inline src/app/shared/components/upload-image/upload-image.component.html]: this one also adding 'on'

File: src/app/shared/components/upload-image/upload-image.component.ts [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+      }
+
+      this.uploadCompleteEmitterService.emit(postAttachments);
+      // clear image preview after upload complete
+      this.imagesToUpload = [];
+    };
+  }
+
+  private getUserInfo (): void {
+    this.accountSettingService.getUserProfile()
```
Comment (caninojories) [inline src/app/shared/components/upload-image/upload-image.component.ts]: get the UseClass in which there is preload of data under shared/classes

File: src/app/shared/components/post-textarea/post-textarea.component.ts [hunk 8]
```diff
@@ -1,14 +1,71 @@
...
+    this.toggleUploadComponent = false;
+  }
+
+  protected showUploadComponent (): void {
+    this.toggleUploadComponent = !this.toggleUploadComponent;
+  }
+
+  /*Destroy subscriber*/
+  public ngOnDestroy (): void {
+    EmitterService.clear(['uploadCompleteEmitter']);
```
Comment (caninojories) [inline src/app/shared/components/post-textarea/post-textarea.component.ts]: How about uploadImagesEmitter?

**Comment by caninojories** [comment]
Merge via  b276e6af95577d785706374fa710d4f81a018434

**Pull request merged**